kernel/qsort.c

```diff
@@ -0,0 +1,266 @@
+/*
```

Comment:
I initially thought there was no precedent to this kind of external inclusion of out-of-kernel-directory sources, but `kernel/Hakefile` references `../usr/drivers/cpuboot/init_ap_x86_64.S`.

Not sure if this constitutes a significant point..